snp_assignment.pl

```diff
@@ -175,6 +175,18 @@ sub _GetVepData {
       		Time::HiRes::sleep($retry);
       		# after sleeping re-request
 		return _GetVepData($id);
+        } elsif($status == 429) {
+                Time::HiRes::sleep(1.0);
+                # after sleeping re-request
+                return _GetVepData($id);
+        } elsif($status == 504) {
```

Comment:
My only question is whether waiting for a single second to retry may not be too short when the issue is with the server being slow. My fear is that this could worsen the issue.
I would suggest increasing the waiting time a bit to a set number of seconds or randomly waiting an amount of time between 1 and 10 seconds.

Reply:
Yup, that sounds like a good idea. Can you add a commit with how to generate a random wait interval that in Perl please? :)

Reply:
Done, I believe this should solve it. However, note that this may not be an optimal solution if it is run thousand of times, as the delay may be too long. Feel free to change it as needed.